stellargraph/mapper/link_mappers.py

```diff
@@ -366,13 +366,6 @@ def get_levels(loc, lsize, samples_per_hop, walks):
         for hns in zip(*head_links):
             node_samples = self.sampler.run(nodes=hns, n=1, n_size=self.num_samples)
 
-            # Isolated nodes will return only themselves in the sample list
```

Comment:
Partly because it breaks a test case where someone wraps a NetworkX DiGraph with StellarGraph. And partly because it is a bad idea to have a special representation in one class, but correct for it in a completely different class. Especially when the correction is simply to put the None's back that were left off.

Reply:
Ok, let's talk more :) Are they not being left off anymore?

Reply:
And yes, the sampler class in data/explorer.py explicitly inserts the None's.

Reply:
At the point of use, if a neighbourhood is empty we still need it full of None's, in order to get the size right. We can therefore either fill it full of None at the point of sampling, or try to reduce space by leaving them off, but at the cost of logic having to reconstruct them later, at a different point in code. Seems easier to do the simple thing, so that the output is always guaranteed to be of the correct size and shape.

Reply:
Makes sense, but let me think about edge cases...

Reply:
BTW: looks like we can always use `SampledHeterogeneousBreadthFirstWalk` instead of `SampledBreadthFirstWalk`? It already does the right thing filling None's in place of missing nodes. In case of homogeneous graphs, there'll only be one node type to loop through.

Reply:
We could - it was interesting to find, after I had included the explicit None's, that 'hetero' did the same thing. I guess the thing to think about (I have no opinion here) is whether we care about the extra overhead involved in repeatedly looking up a single type in a dictionary? If we do, then we keep two classes, one optimised for homogeneous and one not. If we don't, then we could just keep the more general class.